server/src/main/java/org/elasticsearch/search/aggregations/metrics/MinAggregator.java

```diff
@@ -67,7 +92,22 @@ public ScoreMode scoreMode() {
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final LeafBucketCollector sub) throws IOException {
         if (valuesSource == null) {
-            return LeafBucketCollector.NO_OP_COLLECTOR;
+            if (parent == null) {
+                return LeafBucketCollector.NO_OP_COLLECTOR;
+            } else {
+                // we have no parent and the values source is empty so we can skip collecting hits.
+                throw new CollectionTerminatedException();
+            }
+        }
+        if (pointConverter != null) {
+            Number segMin = findLeafMinValue(ctx.reader(), pointField, pointConverter);
+            if (segMin != null) {
+                double min = mins.get(0);
```

Comment:
Could you add a comment here explaining why its ok that we only use bucket ordinal `0`?